operations/src/test/java/edu/unc/lib/boxc/operations/impl/metadata/ExportMetadataCsvServiceTest.java

```diff
@@ -67,25 +75,78 @@ void closeService() throws Exception {
     }
 
     @Test
-    public void metadataCsvTest() throws Exception {
+    public void exportDominoMetadataTest() throws Exception {
+        var collectionRecord = makeRecord(COLLECTION_UUID, ADMIN_UNIT_UUID, ResourceType.Collection,
+                "Collection", null, null, null);
         var workRecord1 = makeWorkRecord(UUID1, "Work 1");
         var workRecord2 = makeWorkRecord(UUID2, "Work 2");
-        mockResults(workRecord1, workRecord2);
-        when(solrSearchService.getSearchResults(any()))
-                .thenReturn(makeResultResponse(workRecord1))
-                .thenReturn(makeResultResponse(workRecord2));
-
-        var resultPath = csvService.exportCsv(asPidList(UUID1, UUID2), agent);
-        var csvRecords = parseCsv(ExportMetadataCsvService.CSV_HEADERS, resultPath);
-        assertContainsEntry(csvRecords, "ref_id", UUID1, "Work 1");
-        assertContainsEntry(csvRecords, "ref_id", UUID2, "Work 2");
+        mockParentResults(collectionRecord);
+        mockChildrenResults(workRecord1, workRecord2);
+
+        var resultPath = csvService.exportCsv(asPidList(COLLECTION_UUID), agent);
+        var csvRecords = parseCsv(ExportDominoMetadataService.CSV_HEADERS, resultPath);
+        assertContainsEntry(csvRecords, UUID1, "ref_id", "Work 1");
+        assertContainsEntry(csvRecords, UUID2, "ref_id", "Work 2");
         assertNumberOfEntries(2, csvRecords);
     }
+    
+    @Test
+    public void recordNotFoundTest() {
+        assertThrows(NullPointerException.class, () -> {
+            when(solrSearchService.getSearchResults(
+                    any()).getResultList());
+            var csvPath = csvService.exportCsv(asPidList(COLLECTION_UUID), agent);
+            assertTrue(Files.notExists(csvPath));
+        });
+    }
+
+    @Test
+    public void userNoPermissionTest() {
+        assertThrows(AccessRestrictionException.class, () -> {
+            doThrow(new AccessRestrictionException())
```

Comment:
Just a safety precaution, but I would generally suggest doing setup like this outside of the assertThrows, otherwise we can have cases where the setup accidental throws an exception (like in the previous test).

Also, the `Files.notExists` assertion is unreachable here again